Can't activate feature with all caps tag via font-feature-settings

litherum opened the pull request:
#### 6e7c8de0a165bc84743986dfccf6175e316a5fc3
<pre>
Can&apos;t activate feature with all caps tag via font-feature-settings
<a href="https://bugs.webkit.org/show_bug.cgi?id=254146">https://bugs.webkit.org/show_bug.cgi?id=254146</a>
rdar://106966601

Reviewed by Tim Nguyen.

The CSS spec[1] says:

&gt; The &lt;string&gt; is a case-sensitive OpenType feature tag.

Chrome and Firefox both treat it this way.

[1] <a href="http://w3c.github.io/csswg-drafts/css-fonts-4/#font-feature-settings-prop">http://w3c.github.io/csswg-drafts/css-fonts-4/#font-feature-settings-prop</a>

Test: fast/text/font-feature-settings-case-sensitive.html

* LayoutTests/fast/text/font-feature-settings-case-sensitive-expected.html: Added.
* LayoutTests/fast/text/font-feature-settings-case-sensitive.html: Added.
* LayoutTests/fast/text/resources/Ahem-feature-x-left-spacing.otf: Added.
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontTag):
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFeatureTagValue):

Canonical link: <a href="https://commits.webkit.org/262463@main">https://commits.webkit.org/262463@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca337e2823c3180e92da404eb012875254b4491b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1584 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1615 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1672 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2505 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/1710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/1563 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1680 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1676 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1490 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1598 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1437 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1434 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2341 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1437 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1420 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1337 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1442 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1460 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2501 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1478 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1323 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1394 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1427 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/392 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1541 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/text/font-feature-settings-case-sensitive.html:
This should be a WPT! There's also potential for adding a mismatch test as a WPT, to mismatch with lowercase "xpos" or uppercase "cv01" (you can have multiple ref/mismatch for a single test)

litherum replied:
Looks like WPT already covers this 👍

litherum committed:
Can't activate feature with all caps tag via font-feature-settings
https://bugs.webkit.org/show_bug.cgi?id=254146
rdar://106966601

Reviewed by Tim Nguyen.

The CSS spec[1] says:

> The <string> is a case-sensitive OpenType feature tag.

Chrome and Firefox both treat it this way.

[1] http://w3c.github.io/csswg-drafts/css-fonts-4/#font-feature-settings-prop

Test: fast/text/font-feature-settings-case-sensitive.html

* LayoutTests/fast/text/font-feature-settings-case-sensitive-expected.html: Added.
* LayoutTests/fast/text/font-feature-settings-case-sensitive.html: Added.
* LayoutTests/fast/text/resources/Ahem-feature-x-left-spacing.otf: Added.
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontTag):
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFeatureTagValue):

Canonical link: https://commits.webkit.org/262463@main
